cider/exec/nextgen/jitlib/llvmjit/LLVMJITControlFlow.cpp

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#include "exec/nextgen/jitlib/llvmjit/LLVMJITControlFlow.h"
+
+#include <llvm/IR/Constants.h>
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/IR/Value.h>
+
+#include "exec/nextgen/jitlib/llvmjit/LLVMJITValue.h"
+
+namespace cider::jitlib {
+
+static llvm::Value* castToBool(llvm::IRBuilder<>& builder, llvm::Value* value) {
+  if (value->getType()->isIntegerTy() && value->getType()->getIntegerBitWidth() != 1) {
+    return builder.CreateICmpNE(value, llvm::ConstantInt::get(value->getType(), 0));
+  } else {
+    return value;
+  }
+}
+
+void LLVMIfBuilder::build(const std::function<JITValuePointer()>& condition,
+                          const std::function<void()>& if_true_block,
```

Comment:
will we provide some utils to create function here in the future? I am not clear how will a complex Expression(like NestedIFControlFlowTest) generate such function automatically.

Reply:
Multi-level nested if could be generated by a recursive function call if it has fixed pattern, just like ```NestedForControlFlowTest``` case shown.